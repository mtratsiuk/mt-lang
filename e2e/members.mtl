(def square
  (|x|
    (* x x)))

(print
  [1 2 42]/2)

(print
  ((([2 nil 4 6]
    /filter Boolean)
    /map square)
    /join "|"))
