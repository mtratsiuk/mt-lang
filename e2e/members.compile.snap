const __mtl_curry = f => (...args) => {
  if (args.length >= f.length) {
    return f(...args);
  }

  return __mtl_curry(f.bind(null, ...args));
};

const square = ((x) => {
  return (x * x);
});

console.log(String([1, 2, 42][2]));

console.log(String([2, null, 4, 6].filter(Boolean).map(square).join("|")));
